src/components/ContributionsEpicReminder.tsx

```diff
@@ -0,0 +1,241 @@
+import React from 'react';
+import { css } from 'emotion';
+import { headline, textSans, body } from '@guardian/src-foundations/typography';
+import { palette, space } from '@guardian/src-foundations';
+import { from } from '@guardian/src-foundations/mq';
+import { ReminderFields } from '../lib/variants';
+import { Lines } from '@guardian/src-ed-lines';
+import { TextInput } from '@guardian/src-text-input';
+import { Button } from '@guardian/src-button';
+import { SvgArrowRightStraight, SvgClose } from '@guardian/src-svgs';
+
+const rootStyles = css`
+    [data-target='success'] {
+        display: none;
+    }
+
+    [data-target='error'] {
+        display: none;
+    }
+
+    &.success [data-target='success'] {
+        display: block;
+    }
+
+    &.success [data-target='form'] {
+        display: none;
+    }
+
+    &.success [data-target='close'] {
+        display: none !important;
+    }
+
+    &.invalid [data-target='input'] {
+        border: 4px solid ${palette.error.main};
+        color: ${palette.error.main};
+    }
```

Comment:
Can you confirm, were you unable to use [TextInput's error functionality](https://www.theguardian.design/2a1e5182b/p/097455-text-input-field/t/18cf92) because it accepts an error message string, and you don't need an error meessage to appear?

Reply:
Nope, that's not the reason. This is an odd piece of work in that we render the React component, then ship the HTML & CSS to the client, along with a piece of vanilla JS to handle the interactions. We don't use React on the client so we can't render/hydrate the component, so we can't get it to re-render with the error message on. We need recreate that state by running a bit of JS on the browser to add/remove some class names.